### PR TITLE
[RHACS] Updated backup restore procedures

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://issues.redhat.com/browse/ROX-13353

Preview:
- Backup https://54563--docspreview.netlify.app/openshift-acs/latest/backup_and_restore/backing-up-acs.html
- Restore https://54563--docspreview.netlify.app/openshift-acs/latest/backup_and_restore/restore-acs.html

Merge into `rhacs-docs` and cherry-pick into:
- `rhacs-docs-4.0`
- `rhacs-docs-4.1`

---

NOTE:

- This PR is for RHACS docs.
- It doesn't require any special labels or milestones.

### CODE REVIEW

File: modules/on-demand-backups-roxctl-admin-pass.adoc [hunk 1]
```diff
@@ -34,6 +40,21 @@ $ roxctl -p <admin_password> -e "$ROX_CENTRAL_ADDRESS" central backup
 ----
 $ roxctl -p <admin_password> -e "$ROX_CENTRAL_ADDRESS" central db backup
 ----
+. Run the `backup` command:
```
Comment (johannes94) [inline modules/on-demand-backups-roxctl-admin-pass.adoc:43]: Looks like the `Run the backup command` section is duplicated in this document. Also in the section above in `on-demand-backups-roxctl-admin-pass.adoc` we document that the backup command is different for version 3.0.54 and older, but in this section and in `on-demand-backups-roxctl-api.adoc` as well, we don't mention that. I think we should be consistent, and either don't mention it at all or mention it in both docs.

**Review by johannes94 — CHANGES_REQUESTED** [review]
Now that I see, how the whole doc looks like in action, I think we'd probably need another section just for this new procedure. IMO we have 2 use cases here:

1. Take a backup of the data stored in ACS & restore that backup 
2. Take a backup of the central cluster resources + the data and restore it

With case 1 being the default for most customers, while case 2 is more like an edge case. Given the changes we have here users who only want to backup & restore the data would be confused because they might not even use the operator install method.

File: backup_and_restore/restore-acs.adoc [hunk 2]
```diff
@@ -8,10 +8,53 @@ toc::[]
...
+[IMPORTANT]
+====
+* Make sure to use the backed-up Central certificates so that secured clusters and API tokens issued by the old Central instance remain valid.
+* If you are deploying to another namespace, you must change the namespace in backed-up resources or commands.
+====
+
+** xref:../backup_and_restore/restore-acs.adoc#restore-central-deployment-roxctl[Restoring Central deployment using the `roxctl` CLI]
+** xref:../backup_and_restore/restore-acs.adoc#restore-deployment-config-operator_restore-acs[Restore Central deployment using the {product-title-short} Operator]
+** xref:../backup_and_restore/restore-acs.adoc#restore-deployment-config-helm_restore-acs[Restore Central deployment using Helm]
+. Follow the section to Restore Central data TODO
```
Comment (gaurav-nelson) [inline backup_and_restore/restore-acs.adoc:59]: @johannes94 Doesn't the previously mentioned sections cover this already?

File: modules/restore-cert-roxctl.adoc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+Doing this allows you to ensure that authentication certificates for Secured clusters and the API tokens remain valid for the restored version.
+If you backed up another instance of {product-title-short} Central, you can use the certificate files from that backup.
+
+[NOTE]
+====
+With the `roxctl` CLI, you can not restore the entire Central deployment. Instead, first you use the `roxctl` CLI to generate new manifests using the certificates in your central data backup. Afterwards, you use those manifests to install Central.
+====
+
+.Prerequisites
+* You must have the `central-tls` backup file.
```
Comment (johannes94) [inline modules/restore-cert-roxctl.adoc:19]: You don't have to have the `central-tls` roxctl cert restore work with the roxctl on-demand backup files.

File: modules/restore-deployment-config-helm.adoc [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+[id="restore-deployment-config-helm_{context}"]
+= Restore Central deployment using Helm
+
+[role="_abstract"]
+You can restore your Central deployment to its original configuration by using Helm.
+To successfully restore, you need the backup of your Central custom resource, `central-tls`, and the administrator password.
+
+.Prerequisites
+* You must have the Helm values backup file.
+* You must have installed the `roxctl` CLI.
```
Comment (johannes94) [inline modules/restore-deployment-config-helm.adoc:14]: You must have the database backup file in order to generate `values-private.yaml`

File: backup_and_restore/restore-acs.adoc (hunk 2 above)
Comment (johannes94) [reply backup_and_restore/restore-acs.adoc:59]: I think it's not, the whole path is in this sequence:

1. Restore deployment (helm or operator method)
2. Restore data (with roxctl no matter which installation method)
3. Restore routing configuration

**Review by johannes94 — CHANGES_REQUESTED** [review]

File: modules/restore-cert-roxctl.adoc [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
```
Comment (johannes94) [inline modules/restore-cert-roxctl.adoc:19]:
```suggestion
* You must have a Red Hat Advanced Cluster Security for Kubernetes backup file.
```
The certificates are also store in the RHACS backup file. When using roxctl, it loads the certificates from this backup file (the database backup). So you don't need the central-tls manifest backup file.

File: backup_and_restore/restore-acs.adoc [hunk 6]
```diff
@@ -8,10 +9,53 @@ toc::[]
...
+include::modules/restore-acs-roxctl-api.adoc[leveloffset=+2]
+
+include::modules/restore-acs-roxctl-admin-pass.adoc[leveloffset=+2]
+
+include::modules/resume-acs-restore.adoc[leveloffset=+2]
+
+[id="restore-central-deployment-roxctl"]
+== Restoring Central deployment using the `roxctl` CLI
+You can restore your Central deployment to its original configuration by using the backups you made.
+To successfully restore, you need Central custom resource, `central-tls` and the administrator password.
```
Comment (johannes94) [inline backup_and_restore/restore-acs.adoc:31]: This is not necessary for roxctl CLI. It extracts this data from the RHACS backup file.

**Review by johannes94 — CHANGES_REQUESTED** [review]
Just a few minor things to change. Good work @gaurav-nelson 👍

File: modules/restore-cert-roxctl.adoc [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+* You must have installed the `roxctl` CLI.
+
+.Procedure
+. Run the interactive install command:
++
+[source,terminal]
+----
+$ roxctl central generate interactive
+----
+. For the following prompt, enter the path of the `central-tls` backup file:
```
Comment (johannes94) [inline modules/restore-cert-roxctl.adoc:29]:
```suggestion
. For the following prompt, enter the path of the {product-title} backup file:
```

**Review by johannes94 — COMMENTED** [review]
Only one small change left from my side 👍

**Review by johannes94 — APPROVED** [review]

File: backup_and_restore/restore-acs.adoc [hunk 8]
```diff
@@ -8,10 +9,51 @@ toc::[]
 [role="_abstract"]
 You can restore {product-title} from an existing backup by using the `roxctl` command-line interface (CLI).
 
-You can use the `roxctl` CLI to restore {product-title} by using the `restore` command. You require either an API token or your administrator password to run this command.
+Depending upon your requirements and the data you have backed up. You can restore from the following types of backups:
 
-include::modules/restore-acs-roxctl-api.adoc[leveloffset=+1]
+. Restore Central database from the Central database backup. Use this to recover from a database failure or data corruption event.
+It allows you to restore and recover the Central database to its earlier functional state.
+. Restore Central from the Central deployment backup. Use this if you are migrating Central to another cluster or namespace. This option restores the configurations of your Central installation.
```
Comment (adellape) [inline backup_and_restore/restore-acs.adoc:16]: Same here re: mix of sentence fragments and whole sentences. Maybe a description list?

```
Restore Central database from the Central database backup:: Use this to recover from a database failure or data corruption event.
It allows you to restore and recover the Central database to its earlier functional state.
Restore Central from the Central deployment backup:: Use this if you are migrating Central to another cluster or namespace. This option restores the configurations of your Central installation.
```

Or to keep the numbering, some mix of bold and a colon:

```
. *Restore Central database from the Central database backup:* Use this to recover from a database failure or data corruption event.
It allows you to restore and recover the Central database to its earlier functional state.
. *Restore Central from the Central deployment backup:* Use this if you are migrating Central to another cluster or namespace. This option restores the configurations of your Central installation.
```

Altho the latter is maybe getting a bit non-standard.

File: backup_and_restore/restore-acs.adoc [hunk 9]
```diff
@@ -8,10 +9,51 @@ toc::[]
...
+
+include::modules/restore-deployment-config-helm.adoc[leveloffset=+1]
+
+[id="restore-central-another-cluster"]
+== Restoring central to another cluster or namespace
+You can use the backups of the {product-title-short} Central database and the deployment to restore Central to another cluster or namespace.
+
+The following list provides a high-level overview of installation steps:
+
+. Depending upon your installation method, you must first restore Central deployment by following the instruction in the following topics:
```
Comment (adellape) [inline backup_and_restore/restore-acs.adoc:47]: "instructions" plural?

File: modules/backup-deployment-config-helm.adoc [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+When you use the Helm chart to install {product-title-short}, you store all the custom configuration for your Central deployment within the custom values that you apply to the Helm chart.
+
+You can backup the custom values and save it in a YAML file.
+
+.Procedure
+* Run the following command to backup custom Helm chart values in a YAML file:
++
+[source,terminal]
+----
+$ helm get values --all -n <central-namespace> <central-helm-release> -o yaml > central-values-backup.yaml
```
Comment (adellape) [inline modules/backup-deployment-config-helm.adoc:18]: Underscores for the replaceables.

Looks like there are more throughout the PR, so those too.

File: modules/backup-deployment-config-operator.adoc [hunk 11]
```diff
@@ -0,0 +1,30 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/backing-up-acs.adoc
+:_content-type: PROCEDURE
+[id="backup-deployment-config-operator_{context}"]
+= Backup deployment using the {product-title-short} Operator
+
+[role="_abstract"]
+When you use the {product-title-short} Operator to instal {product-title-short}, {ocp} stores all the custom configuration for your Central deployment within the Central custom resource.
+You can backup the Central custom resource, `central-tls` and the administrator password. `central-tls` includes the certificates for authenticating with Secured clusters and signing API tokens.
```
Comment (adellape) [inline modules/backup-deployment-config-operator.adoc:10]: I think another comma is needed in the list? Altho I wasn't sure if `central-tls` is the `Secret` name or the "backup file" as it's referred to elsewhere. Describing a bit more what `central-tls` is here might help. E.g.:

```suggestion
You can backup the Central custom resource, the `central-tls` secret, and the administrator password.  The `central-tls` secret includes the certificates for authenticating with Secured clusters and signing API tokens.
```

Or `s/secret/backup file/` if that's more accurate at this stage.

File: modules/restore-cert-roxctl.adoc [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+----
+$ roxctl central generate interactive
+----
+. For the following prompt, enter the path of the {product-title} backup file:
++
+[source,terminal]
+----
+Enter path to the backup bundle from which to restore keys and certificates (optional): <backup-file-path>
+----
+. For other following prompts, press *Enter* to accept the default value or enter custom values as required.
```
Comment (adellape) [inline modules/restore-cert-roxctl.adoc:35]: Not important, but:

`kbd:[Enter]` will render a fun "Enter" button 🙃

File: modules/restore-deployment-config-helm.adoc [hunk 13]
```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/restore-acs.adoc
+:_content-type: PROCEDURE
+[id="restore-deployment-config-helm_{context}"]
+= Restore Central deployment using Helm
+
+[role="_abstract"]
+You can restore your Central deployment to its original configuration by using Helm.
+To successfully restore, you need the backup of your Central custom resource, `central-tls`, and the administrator password.
```
Comment (adellape) [inline modules/restore-deployment-config-helm.adoc:10]: Same re: describing what `central-tls` is here (and one more instance below).

File: modules/restore-deployment-config-helm.adoc [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+* You must have installed the `roxctl` CLI.
+
+.Procedure
+. Generate `values-private.yaml`` from the {product-title-short} database backup file:
++
+[source,terminal]
+----
+$ roxctl central generate k8s pvc --backup-bundle <path-to-backup-file> --output-format "helm-values"
+----
+. Run the `helm install` command and specify you backup files:
```
Comment (adellape) [inline modules/restore-deployment-config-helm.adoc:24]:
```suggestion
. Run the `helm install` command and specify your backup files:
```

File: modules/restore-deployment-config-helm.adoc [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+You can restore your Central deployment to its original configuration by using Helm.
+To successfully restore, you need the backup of your Central custom resource, `central-tls`, and the administrator password.
+
+.Prerequisites
+* You must have the Helm values backup file.
+* You must have a {product-title} backup file.
+* You must have installed the `roxctl` CLI.
+
+.Procedure
+. Generate `values-private.yaml`` from the {product-title-short} database backup file:
```
Comment (adellape) [inline modules/restore-deployment-config-helm.adoc:18]:
```suggestion
. Generate `values-private.yaml` from the {product-title-short} database backup file:
```

File: backup_and_restore/restore-acs.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (gaurav-nelson) [reply backup_and_restore/restore-acs.adoc]: That was outdated content, not using it anymore.

**Pull request merged**